x/emissions/migrations/v3/migrate_test.go

```diff
@@ -0,0 +1,72 @@
+package v3_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	keepertest "github.com/zeta-chain/zetacore/testutil/keeper"
+	"github.com/zeta-chain/zetacore/x/emissions/exported"
+	v3 "github.com/zeta-chain/zetacore/x/emissions/migrations/v3"
+	"github.com/zeta-chain/zetacore/x/emissions/types"
+)
+
+type mockSubspace struct {
+	ps types.Params
+}
+
+func newMockSubspace(ps types.Params) mockSubspace {
+	return mockSubspace{ps: ps}
+}
+
+func (ms mockSubspace) GetParamSet(ctx sdk.Context, ps exported.ParamSet) {
+	*ps.(*types.Params) = ms.ps
+}
+
+func TestMigrate(t *testing.T) {
```

Comment:
NIT : Can we also add a test in which some values are missing in `legacyParams` .